runtime/criusupport/criusupport.cpp

```diff
@@ -347,8 +351,143 @@ releaseSafeOrExcusiveVMAccess(J9VMThread *currentThread, J9InternalVMFunctions *
 	}
 }
 
+static char*
+endLine(char *cursor)
+{
+	while(true) {
+		switch (*cursor) {
+		case '\r':
+		case '\n':
+		case '\0':
+			if ((cursor[0] == '\r' && cursor[1] == '\n')
```

Comment:
How do we know that `cursor[1]` is valid to look at and not past the end of our buffer?  If we have a requirement of a null terminated buffer, then we should state that or any other contracts in the comment for the function.

Reply:
The file buffer is allocated with an extra byte which is set to '\0'